x-pack/plugins/ml/server/lib/alerts/alerting_service.ts

```diff
@@ -65,18 +71,53 @@ const resultTypeScoreMapping = {
 export function alertingServiceProvider(
   mlClient: MlClient,
   datafeedsService: DatafeedsService,
-  getFieldsFormatRegistry: FieldFormatsRegistryProvider
+  getFieldsFormatRegistry: FieldFormatsRegistryProvider,
+  getDataViewsService: GetDataViewsService
 ) {
   type FieldFormatters = AwaitReturnType<ReturnType<typeof getFormatters>>;
 
-  const getFormatters = memoize(async () => {
+  /**
+   * Provides formatters based on the data view of the datafeed index pattern
+   * and set of default formatters for fallback.
+   */
+  const getFormatters = memoize(async (datafeed: MlDatafeed) => {
     const fieldFormatsRegistry = await getFieldsFormatRegistry();
     const numberFormatter = fieldFormatsRegistry.deserialize({ id: FIELD_FORMAT_IDS.NUMBER });
+
+    const fieldFormatMap = await getFieldsFormatMap(datafeed.indices[0]);
+
+    const fieldFormatters = fieldFormatMap
+      ? Object.entries(fieldFormatMap).reduce((acc, [fieldName, config]) => {
+          const formatter = fieldFormatsRegistry.deserialize(config);
+          acc[fieldName] = formatter.convert.bind(formatter);
+          return acc;
+        }, {} as Record<string, IFieldFormat['convert']>)
+      : {};
+
     return {
       numberFormatter: numberFormatter.convert.bind(numberFormatter),
+      fieldFormatters,
     };
   });
 
+  /**
+   * Attempts to find a data view based on the index pattern
+   */
+  const getFieldsFormatMap = memoize(
+    async (indexPattern: string): Promise<Record<string, SerializedFieldFormat> | undefined> => {
+      try {
+        const dataViewsService = await getDataViewsService();
+
+        const dataView = (await dataViewsService.find(indexPattern, 1))[0];
```

Comment:
if this `indexPattern` string contains a wildcard you can't guarantee that the first found data view is the correct one.
performing a find to match the exact name is safer